drivers/hmis/app/models/hmis/ce/match/internal/candidate_repository.rb

```diff
@@ -18,10 +18,18 @@ def import_proxies(clients, timestamp:)
           client_type: client.class.sti_name,
           created_at: timestamp,
           updated_at: timestamp,
+          deleted_at: nil,
         }
       end
 
-      result = Hmis::Ce::ClientProxy.import(values, on_duplicate_key_ignore: true)
+      result = Hmis::Ce::ClientProxy.import(
+        values,
+        on_duplicate_key_update: {
+          conflict_target: [:client_type, :client_id],
+          columns: [:deleted_at, :updated_at],
+          condition: 'ce_client_proxies.deleted_at IS NOT NULL',
```

Comment:
The update happens only if the existing row is soft-deleted

- If no row exists: a new proxy is inserted.
- If a soft-deleted row exists: it is “revived” by setting deleted_at to NULL (from the insert values) and updating updated_at.
- If an active row exists: the WHERE condition fails, so no UPDATE occurs (the upsert degrades to DO NOTHING), leaving updated_at unchanged.